### PR TITLE
qa/tasks: incorrect exception in the modules test

### PR DESCRIPTION
**Pull request by nizamial09**
## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "quincy"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

- When filling out the below checklist, you may click boxes directly in the GitHub web UI.  When entering or editing the entire PR message in the GitHub web UI editor, you may also select a checklist item by adding an `x` between the brackets: `[x]`.  Spaces and capitalization matter when checking off items this way.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
- `jenkins test rook e2e`
</details>

### CODE REVIEW

**Review by epuertat — APPROVED** [review]

**Comment by epuertat** [comment]
jenkins retest this please

File: qa/tasks/mgr/dashboard/test_mgr_module.py [hunk 1]
```diff
@@ -15,7 +15,7 @@
 class MgrModuleTestCase(DashboardTestCase):
     MGRS_REQUIRED = 1
 
-    @retry(on_exception=RuntimeError, tries=2, delay=0.5, logger=logger)
+    @retry(on_exception=requests.ConnectionError, tries=2, delay=0.5, logger=logger)
```
[the remark this answers is not in the document]
Comment (nizamial09) [reply qa/tasks/mgr/dashboard/test_mgr_module.py]: thanks, did that.

**Comment by nizamial09** [comment]
jenkins test api

**Comment by nizamial09** [comment]
jenkins test api

**Comment by nizamial09** [comment]
looks like this is not properly fixing the issue. so could it be that the proper way to handle this is what @NitzanMordhai [suggested here](https://tracker.ceph.com/issues/62972#note-25)? can you have a look? @epuertat 

```
2024-09-18 03:11:13,299.299 INFO:__main__:test_list_enabled_module (tasks.mgr.dashboard.test_mgr_module.MgrModuleTest) ... ERROR
2024-09-18 03:11:13,299.299 INFO:__main__:
2024-09-18 03:11:13,300.300 INFO:__main__:Stopped test: test_list_enabled_module (tasks.mgr.dashboard.test_mgr_module.MgrModuleTest) in 6.116805s
2024-09-18 03:11:13,300.300 INFO:__main__:
2024-09-18 03:11:13,300.300 INFO:__main__:======================================================================
2024-09-18 03:11:13,300.300 INFO:__main__:ERROR: test_list_enabled_module (tasks.mgr.dashboard.test_mgr_module.MgrModuleTest)
2024-09-18 03:11:13,300.300 INFO:__main__:----------------------------------------------------------------------
2024-09-18 03:11:13,300.300 INFO:__main__:Traceback (most recent call last):
2024-09-18 03:11:13,300.300 INFO:__main__:  File "/tmp/tmp.0CiUKpjz7z/venv/lib/python3.10/site-packages/urllib3/connection.py", line 199, in _new_conn
2024-09-18 03:11:13,300.300 INFO:__main__:    sock = connection.create_connection(
2024-09-18 03:11:13,300.300 INFO:__main__:  File "/tmp/tmp.0CiUKpjz7z/venv/lib/python3.10/site-packages/urllib3/util/connection.py", line 85, in create_connection
2024-09-18 03:11:13,300.300 INFO:__main__:    raise err
2024-09-18 03:11:13,300.300 INFO:__main__:  File "/tmp/tmp.0CiUKpjz7z/venv/lib/python3.10/site-packages/urllib3/util/connection.py", line 73, in create_connection
2024-09-18 03:11:13,300.300 INFO:__main__:    sock.connect(sa)
2024-09-18 03:11:13,300.300 INFO:__main__:  File "/tmp/tmp.0CiUKpjz7z/venv/lib/python3.10/site-packages/gevent/_socketcommon.py", line 590, in connect
2024-09-18 03:11:13,300.300 INFO:__main__:    self._internal_connect(address)
2024-09-18 03:11:13,300.300 INFO:__main__:  File "/tmp/tmp.0CiUKpjz7z/venv/lib/python3.10/site-packages/gevent/_socketcommon.py", line 634, in _internal_connect
2024-09-18 03:11:13,300.300 INFO:__main__:    raise _SocketError(err, strerror(err))
2024-09-18 03:11:13,300.300 INFO:__main__:ConnectionRefusedError: [Errno 111] Connection refused
2024-09-18 03:11:13,300.300 INFO:__main__:
2024-09-18 03:11:13,300.300 INFO:__main__:The above exception was the direct cause of the following exception:
2024-09-18 03:11:13,300.300 INFO:__main__:
2024-09-18 03:11:13,300.300 INFO:__main__:Traceback (most recent call last):
2024-09-18 03:11:13,300.300 INFO:__main__:  File "/tmp/tmp.0CiUKpjz7z/venv/lib/python3.10/site-packages/urllib3/connectionpool.py", line 789, in urlopen
2024-09-18 03:11:13,300.300 INFO:__main__:    response = self._make_request(
2024-09-18 03:11:13,300.300 INFO:__main__:  File "/tmp/tmp.0CiUKpjz7z/venv/lib/python3.10/site-packages/urllib3/connectionpool.py", line 490, in _make_request
2024-09-18 03:11:13,300.300 INFO:__main__:    raise new_e
2024-09-18 03:11:13,300.300 INFO:__main__:  File "/tmp/tmp.0CiUKpjz7z/venv/lib/python3.10/site-packages/urllib3/connectionpool.py", line 466, in _make_request
2024-09-18 03:11:13,301.301 INFO:__main__:    self._validate_conn(conn)
2024-09-18 03:11:13,301.301 INFO:__main__:  File "/tmp/tmp.0CiUKpjz7z/venv/lib/python3.10/site-packages/urllib3/connectionpool.py", line 1095, in _validate_conn
2024-09-18 03:11:13,301.301 INFO:__main__:    conn.connect()
2024-09-18 03:11:13,301.301 INFO:__main__:  File "/tmp/tmp.0CiUKpjz7z/venv/lib/python3.10/site-packages/urllib3/connection.py", line 693, in connect
2024-09-18 03:11:13,301.301 INFO:__main__:    self.sock = sock = self._new_conn()
2024-09-18 03:11:13,301.301 INFO:__main__:  File "/tmp/tmp.0CiUKpjz7z/venv/lib/python3.10/site-packages/urllib3/connection.py", line 214, in _new_conn
2024-09-18 03:11:13,301.301 INFO:__main__:    raise NewConnectionError(
2024-09-18 03:11:13,301.301 INFO:__main__:urllib3.exceptions.NewConnectionError: <urllib3.connection.HTTPSConnection object at 0x7f6773dbe9e0>: Failed to establish a new connection: [Errno 111] Connection refused
2024-09-18 03:11:13,301.301 INFO:__main__:
2024-09-18 03:11:13,301.301 INFO:__main__:The above exception was the direct cause of the following exception:
2024-09-18 03:11:13,301.301 INFO:__main__:
2024-09-18 03:11:13,301.301 INFO:__main__:Traceback (most recent call last):
2024-09-18 03:11:13,301.301 INFO:__main__:  File "/tmp/tmp.0CiUKpjz7z/venv/lib/python3.10/site-packages/requests/adapters.py", line 667, in send
2024-09-18 03:11:13,301.301 INFO:__main__:    resp = conn.urlopen(
2024-09-18 03:11:13,301.301 INFO:__main__:  File "/tmp/tmp.0CiUKpjz7z/venv/lib/python3.10/site-packages/urllib3/connectionpool.py", line 843, in urlopen
2024-09-18 03:11:13,301.301 INFO:__main__:    retries = retries.increment(
2024-09-18 03:11:13,301.301 INFO:__main__:  File "/tmp/tmp.0CiUKpjz7z/venv/lib/python3.10/site-packages/urllib3/util/retry.py", line 519, in increment
2024-09-18 03:11:13,301.301 INFO:__main__:    raise MaxRetryError(_pool, url, reason) from reason  # type: ignore[arg-type]
2024-09-18 03:11:13,301.301 INFO:__main__:urllib3.exceptions.MaxRetryError: HTTPSConnectionPool(host='172.21.2.17', port=7790): Max retries exceeded with url: /api/mgr/module (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x7f6773dbe9e0>: Failed to establish a new connection: [Errno 111] Connection refused'))
2024-09-18 03:11:13,301.301 INFO:__main__:
2024-09-18 03:11:13,301.301 INFO:__main__:During handling of the above exception, another exception occurred:
2024-09-18 03:11:13,301.301 INFO:__main__:
2024-09-18 03:11:13,301.301 INFO:__main__:Traceback (most recent call last):
2024-09-18 03:11:13,301.301 INFO:__main__:  File "/home/jenkins-build/build/workspace/ceph-api/qa/tasks/mgr/dashboard/test_mgr_module.py", line 61, in test_list_enabled_module
2024-09-18 03:11:13,301.301 INFO:__main__:    data = self._get('/api/mgr/module')
2024-09-18 03:11:13,301.301 INFO:__main__:  File "/home/jenkins-build/build/workspace/ceph-api/qa/tasks/mgr/dashboard/helper.py", line 342, in _get
2024-09-18 03:11:13,301.301 INFO:__main__:    return cls._request(url, 'GET', params=params, version=version,
2024-09-18 03:11:13,301.301 INFO:__main__:  File "/home/jenkins-build/build/workspace/ceph-api/qa/tasks/mgr/dashboard/helper.py", line 314, in _request
2024-09-18 03:11:13,301.301 INFO:__main__:    cls._resp = cls._session.get(url, params=params, verify=False,
2024-09-18 03:11:13,301.301 INFO:__main__:  File "/tmp/tmp.0CiUKpjz7z/venv/lib/python3.10/site-packages/requests/sessions.py", line 602, in get
2024-09-18 03:11:13,302.302 INFO:__main__:    return self.request("GET", url, **kwargs)
2024-09-18 03:11:13,302.302 INFO:__main__:  File "/tmp/tmp.0CiUKpjz7z/venv/lib/python3.10/site-packages/requests/sessions.py", line 589, in request
2024-09-18 03:11:13,302.302 INFO:__main__:    resp = self.send(prep, **send_kwargs)
2024-09-18 03:11:13,302.302 INFO:__main__:  File "/tmp/tmp.0CiUKpjz7z/venv/lib/python3.10/site-packages/requests/sessions.py", line 703, in send
2024-09-18 03:11:13,302.302 INFO:__main__:    r = adapter.send(request, **kwargs)
2024-09-18 03:11:13,302.302 INFO:__main__:  File "/tmp/tmp.0CiUKpjz7z/venv/lib/python3.10/site-packages/requests/adapters.py", line 700, in send
2024-09-18 03:11:13,302.302 INFO:__main__:    raise ConnectionError(e, request=request)
2024-09-18 03:11:13,302.302 INFO:__main__:requests.exceptions.ConnectionError: HTTPSConnectionPool(host='172.21.2.17', port=7790): Max retries exceeded with url: /api/mgr/module (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x7f6773dbe9e0>: Failed to establish a new connection: [Errno 111] Connection refused'))
2024-09-18 03:11:13,302.302 INFO:__main__:
```

**Comment by epuertat** [comment]
> looks like this is not properly fixing the issue. so could it be that the proper way to handle this is what @NitzanMordhai [suggested here](https://tracker.ceph.com/issues/62972#note-25)? can you have a look? @epuertat


@nizamial09 I left [a comment in there](https://tracker.ceph.com/issues/62972#note-43).

**Pull request closed without merging**